### PR TITLE
Reverse commit order between assets file and targets

### PR DESCRIPTION
**Pull request by emgarten**
Commit targets before the assets file.

//cc @joelverhagen @alpaix @rrelyea @jainaashish  @mishra14

### CODE REVIEW

**Comment by joelverhagen** [comment]
What issue does this fix?

File: src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs [hunk 1]
```diff
@@ -124,6 +124,8 @@ public ISet<LibraryRange> GetAllUnresolved()
 
             var isTool = OutputType == RestoreOutputType.DotnetCliTool;
 
+            MSBuild.Commit(log);
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs]: Add a comment explaining why this particular order is important.

Comment (rrelyea) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs]: fixes https://github.com/NuGet/Home/issues/3877

Comment (emgarten) [reply src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs]: sure

**Review by alpaix — APPROVED** [review]

**Review by joelverhagen — APPROVED** [review]

**Pull request merged**